test/spark.js

```diff
@@ -363,3 +363,60 @@ test('calculateDelayBeforeNextTask() handles one task per round', () => {
   })
   assertEquals(delay, 60_000)
 })
+
+test('calculateDelayBeforeNextTask() introduces random jitter', () => {
+  const getDelay = () => calculateDelayBeforeNextTask({
+    lastTaskDurationInMs: 3_000,
+
+    // one task every 10 seconds (on average)
+    roundLengthInMs: 60_000,
+    maxTasksPerRound: 6,
+
+    // jitter up to 1 second
+    maxJitterInMs: 1_000
+  })
+
+  const delay1 = getDelay()
+  const delay2 = getDelay()
+
+  assertGreaterOrEqual(delay1, 7_000)
+  assertLessOrEqual(delay1, 7_000 + 1_000)
+
+  assertNotEquals(
+    delay1,
+    delay2,
+    `Expected delay values to be different because of jitter. Actual value: ${delay1}`
+  )
```

Comment:
This fails if `Math.random()` produces the same value twice. It's very very unlikely to happen, I'm just documenting it here for completeness

Reply:
Should we maybe mock `Math.random`? It would make tests deterministic.

Reply:
I feel mocking Math.random is not worth the effort right now. We can re-run the test if it fails, the second run is even more unlikely to fail.

I am happy to revisit this decision if these new tests start to fail regularly.